packages/cli/src/lib/bundler/config.ts

```diff
@@ -70,27 +70,74 @@ async function readBuildInfo() {
   };
 }
 
-async function loadLernaPackages(): Promise<
-  { name: string; location: string }[]
-> {
+type LernaPackage = {
+  name: string;
+  location: string;
+};
+
+async function loadLernaPackages(): Promise<LernaPackage[]> {
   const LernaProject = require('@lerna/project');
   const project = new LernaProject(cliPaths.targetDir);
   return project.getPackages();
 }
 
+// Enables proper resolution of packages when linking in external packages.
+// Without this the packages would depend on dependencies in the node_modules
+// of the external packages themselves, leading to module duplication
+class LinkedPackageResolvePlugin implements ResolvePlugin {
+  constructor(
+    private readonly targetModules: string,
+    private readonly packages: LernaPackage[],
+  ) {}
+
+  apply(resolver: any /* EnhancedResolve.Resolver */) {
```

Comment:
Assume all the `any` in here were necessary

Reply:
Yeah was having a look for proper types but not sure it's worth it, the ones from webpack are a bit incomplete

Reply:
We're on webpack 4 which uses `enhanced-resolve` 4. There's a separate type package for 3, and types shipped in 5. Just sticking to `any` with a few manual types xP